src/swaps/lifi/LifiSwapProvider.ts

```diff
@@ -0,0 +1,321 @@
+  QuoteRequest,
+  SwapRequest,
+  SwapStatus,
+} from '../types';
+
+export interface LifiSwapProviderConfig extends EvmSwapProviderConfig {
+  rpcURL: string;
+}
+
+export interface LifiSwapHistoryItem extends EvmSwapHistoryItem {
+  fromFundHash: string;
+  fromFundTx: Transaction<EvmTypes.EthersTransactionResponse>;
+}
+
+class LifiSwapProvider extends EvmSwapProvider {
+  readonly config: LifiSwapProviderConfig;
+  readonly _lifiClient: LiFi;
+  private _httpClient: HttpClient;
+  readonly nativeAssetAddress: string = '0x0000000000000000000000000000000000000000';
+
+  constructor(config: LifiSwapProviderConfig) {
+    super(config);
+    this._lifiClient = new LiFi();
+    this._httpClient = new HttpClient({ baseURL: this.config.rpcURL });
+  }
+
+  async getSupportedPairs() {
+    return [];
+  }
+
+  // returns rates between tokens
+  async getQuote({ network, from, to, amount, fromAccountId, toAccountId, walletId }: QuoteRequest) {
+    const fromInfo = cryptoassets[from];
+    const toInfo = cryptoassets[to];
+
+    const fromAmountInUnit = currencyToUnit(fromInfo, new BN(amount)).toFixed();
+
+    const fromChainId = ChainNetworks[fromInfo.chain][network].chainId;
+    const toChainId = ChainNetworks[toInfo.chain][network].chainId;
+
+    const fromAddressRaw = await this.getSwapAddress(network, walletId as string, from, fromAccountId as string);
+    const toAddressRaw = await this.getSwapAddress(network, walletId as string, to, toAccountId as string);
+
+    const quoteRequest = {
+      fromChain: fromChainId as number,
+      fromAmount: fromAmountInUnit,
+      fromToken: fromInfo.contractAddress ?? this.nativeAssetAddress,
+      toChain: toChainId as number,
+      toToken: toInfo.contractAddress ?? this.nativeAssetAddress,
+      fromAddress: chains[fromInfo.chain].formatAddress(fromAddressRaw),
+      toAddress: chains[toInfo.chain].formatAddress(toAddressRaw),
+      ...this.getDefaultRouteOptions(),
+    };
+
+    try {
+      const bestRoute = await this._lifiClient.getQuote(quoteRequest);
+      return { from, to, fromAmount: fromAmountInUnit, toAmount: bestRoute.estimate.toAmount, bestRoute: bestRoute };
+    } catch (e) {
+      console.warn(e);
+      return null;
+    }
+  }
+
+  async newSwap(swap: SwapRequest<LifiSwapHistoryItem>) {
+    const approvalAddress = swap.quote.bestRoute?.estimate.approvalAddress;
+    const approveTx = await this.approve(swap, false, approvalAddress);
+    const updates = approveTx || (await this.initiateSwap(swap));
+
+    return { id: uuidv4(), ...updates };
+  }
+
+  private async initiateSwap({ network, walletId, quote }: SwapRequest<LifiSwapHistoryItem>) {
+    const client = super.getClient(network, walletId, quote.from, quote.fromAccountId);
+    const txData: TransactionRequest = quote.bestRoute?.transactionRequest as TransactionRequest;
+    const fromFundTx = await client.wallet.sendTransaction(txData);
+
+    return {
+      status: 'WAITING_FOR_INITIATION_CONFIRMATIONS',
```

Comment:
PLS, let's use the PR to introduce enums for the statuses

Reply:
This will be resolved in other PR.